build.gradle

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * This generated file contains a sample Java project to get you started.
+ * For more details take a look at the Java Quickstart chapter in the Gradle
+ * User Manual available at https://docs.gradle.org/5.2.1/userguide/tutorial_java_projects.html
+ */
+
+plugins {
+    // Apply the java plugin to add support for Java
+    id 'java'
+    // git clone, etc. 
+    id "org.ajoberstar.grgit" version "2.3.0"       
+}
+
+sourceCompatibility = '1.8'
+targetCompatibility = '1.8'
+version = '0.1.0'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    // Use JUnit test framework
+    testCompile "junit:junit-dep:4.11"
+    // com.googlecode.json.simple
+    compile "com.googlecode.json-simple:json-simple:1.1.1"
+    // org.apache.commons.math3
+    compile "org.apache.commons:commons-math3:3.6.1"
+    // gov.nist.math.jama
+    compile "gov.nist.math:jama:1.0.2"
+}
+
+// method to add/update an external project from git and add it to the 
+// sourceset
+ext.addExternalProject = { path, url, srcDir ->
+  File projDir = new File("$path")
+  // clone project if it doesn't exist locally
+  if(!projDir.exists()) {
+    grgit.clone(dir: projDir, uri: url)
+  }
+  // pull the project if it does exist locally
+  else {
+    grgit.open(dir: projDir)
+    grgit.pull()
+    grgit.close()
+  }
+
+  // add to sourceset
+  sourceSets {
+    main {
+      java {
+          srcDirs "$path/$srcDir"
```

Comment:
should this be `srcDirs += "$path/$srcDir"`?

Reply:
Apparently not? since it builds? and based on: https://stackoverflow.com/questions/31077844/add-another-java-source-directory-to-gradle-script